dataP

ashbeekim opened the pull request:
# EDA/FE
* 목적 : 각기 다른 데이터 전처리 및 엔지니어링과 무관하게 @qkrwjdduf159 의 모델 적용
* 사용방법
```
data = DHL()
data.train    # train 데이터 확인
data.test.    # test 데이터 확인

data.groupby_mean().   # code성 데이터를 적용한 추가된 변수를 mean값으로 매핑
data.control_params({
    "drop_column": [],        # 최종 학습 전, 제거할 변수
    "dummy_column": []    # 최종 학습 전, 더미화 할 변수
})

data.f_train    # 최종 train 데이터 확인
data.f_test.   # 최종 test 데이터 확인
```

ashbeekim committed:
Create dataP.py

- 정열님 모델(stacking으로 작성됨) 데이터 처리
- 더미 변수 및 제거 변수의 유동적인 처리

ashbeekim committed:
Update dataP.py

* 모듈 정의 : 사용해야 할 주 변수는 달라지지 않고 현재까지의 점수 중 높은 점수를 받은 @qkrwjdduf159 (공동팀장 중 1인)의 모델 적용에 최적화된 데이터 처리 모듈
* Notes:
 - train 및 test 데이터가 이미 train_DF,  test_DF로 선언이 되어 있다면, 따로 작성해야 할 부분이 발생하지 않음
 - D, H, L 코드 데이터는 현재 위치 기반으로 불러올 수 있도록 작성했음

ashbeekim committed:
Update dataP.py

co-author : @qkrwjdduf159 , @oliviachchoi, @AshbeeKim 
contributor : @kim-hyun-ho

AshbeeKim commented:
[Jobcare.ipynb](https://colab.research.google.com/drive/1ZpIE8vPDFeBI8CTUDMDodGUOJf_8Hfnt?usp=sharing)

요즘 개인적으로는 ipynb 업로드를 지양하고 있어서, 혹시 내용 확인이 필요하시면 링크 클릭해주세요.
확인 요청 시, 공유 권한 드리겠습니다. 목차 mixed>data process를 확인하시면 됩니다~

qkrwjdduf159 commented:
데이터 로직에 대해서 쉽게 이해가 되지 않아서 혹시 간단한 설명을 적어주실 수 있나요?

ashbeekim committed:
Update dataP.py

Request review @qkrwjdduf159

qkrwjdduf159 commented on dataP.py:
```diff
@@ -0,0 +1,105 @@
+    class 내 property는 def별 활용이 높거나, 기본 변수를 해하지 않도록 선언
+    train, test의 경우, 기본 제공 데이터에서 EDA 중 변수가 늘어날 수 있기에 함수 내 불러올 데이터만 조정하면 되도록 작성
+    '''
+    code_list = []
+    for csv in glob("./*.csv"):
+        if re.sub("[ㄱ-힣]", "", csv)[2]=="_":
+            code_list.append(csv)
+    for csv in code_list:
+        _n = (re.sub("[^a-zA-Z]", "", csv.split(".")[1])).lower()
+        locals()[f"{_n}_df"] = pd.read_csv(csv, encoding="utf-8", index_col=0).T.to_dict()
+
+    train = eval("train_DF")
+    test = eval("test_DF")
```
지정 경로를 위한 파라미터가 하나 생긴다면 코랩 사용자에게 편리할 것 같습니다.

AshbeeKim replied:
> 지정 경로를 위한 파라미터가 하나 생긴다면 코랩 사용자에게 편리할 것 같습니다.

를 반영해서 새로 commit 했습니다~

ashbeekim committed:
Update dataP.py

입력받은 경로 기준으로 작동
train, test 데이터는 전처리가 다를 수 있기 때문에 이전의 "train_DF", "test_DF"로 선언되어야 하는 부분 수정하지 않았음

ashbeekim committed:
Update dataP.py

- train, test를 init에서 받아와서 처리하도록 변경함.

Commit:
Updated dataP.py to modules/dataP.py

- 상속 관계의 클래스 구성으로 보다 더 유연한 데이터 처리가 가능하도록 작성

Commit:
Create main.py

- Tutorial to use modules/dataP/DHL
- For Colab user(who is using .ipynb)
```
%load_ext autoreload
%autoreload 2
from modules.dataP import DHL
data = DHL()
```

AshbeeKim commented on modules/dataP.py:
```diff
@@ -81,7 +109,7 @@ def convert_code(self, datas):
+            comp = self.dicts[f"{_t}_df"]
             DF[_a] = DF[_d].apply(lambda x: comp[x][_r])
 
         if datas == "train":
```
이후 진행되는 함수가 mean으로 처리되는 값인데, mean 말고 다른 방법을 사용해보라는 의견도 있었기에 다음의 코드를 진행했습니다.

```
for col in data.match_cols[:4]:
    print(f"\n{col}:\n\tmean : {str((data.train[col]).mean())}\n\tmedian : {str((data.train[col]).median())}\n\tmode : {str((data.train[col].apply(lambda x: str(round(x, 3)))).mode())}")
for col in data.match_cols[-4:]:
    print(f"\n{col}:\n\tmean : {str((data.train[col]).mean())}\n\tmedian : {str((data.train[col]).median())}\n\tmode : {str((data.train[col].apply(lambda x: str(round(x, 3)))).mode())}")
```
의 결과는 아래와 같습니다.

```
person_prefer_h_1_m:
	mean : 409.64880237314003
	median : 398.0
	mode : 0    316
dtype: object

contents_attribute_h_m:
	mean : 422.4504901872892
	median : 407.0
	mode : 0    543
dtype: object

person_prefer_h_1_l:
	mean : 107.44251132082614
	median : 94.0
	mode : 0    94
dtype: object

contents_attribute_h_l:
	mean : 118.56438377451185
	median : 94.0
	mode : 0    94
dtype: object

person_prefer_d_1_l:
	mean : 421.91670501702356
	median : 377.0
	mode : 0    1
dtype: object

person_prefer_d_2_l:
	mean : 425.7630187010286
	median : 377.0
	mode : 0    1
dtype: object

person_prefer_d_3_l:
	mean : 412.1727061007947
	median : 377.0
	mode : 0    1
dtype: object

contents_attribute_d_l:
	mean : 460.8365238838054
	median : 522.0
	mode : 0    1
dtype: object
```
위와 같은 경우, median으로 시도해 볼 가치가 있나요?
아니면 벌써 시도한 분이 계실까요?

qkrwjdduf159 replied:
그런데 target을 가지고 mean을 찍어서 나온 결과물로 진행을 했었는데 생각을 다시 해보니 mean말고 median으로 찍게 된다면 1과 0 사이의 변수만 가지고 median을 찍을 때 1과 0중 더 많은 부분을 택할 것 같아서 못했습니다. 그러나 다른 변수에서는 고려해보려 합니다!!

yunyoseob committed:
Update README.md
